cluster-autoscaler/cloudprovider/azure/azure_cache.go

```diff
@@ -353,6 +359,45 @@ func (m *azureCache) getAutoscalingOptions(ref azureRef) map[string]string {
 	return m.autoscalingOptions[ref]
 }
 
+// HasInstance returns if a given instance exists in the azure cache
+func (m *azureCache) HasInstance(instance *azureRef) (bool, error) {
+	vmsPoolSet := m.getVMsPoolSet()
```

Comment:
Why is this only applicable to `VMsPoolSet`?

Reply:
I am uncertain what this question means. Can you elaborate?

Reply:
Yep, so it's unclear to me why this function is only checking VMsPool and not VMSS, as well.

Maybe I'm misunderstanding how these two nodegroup types interact? But my understanding is that `getVMsPoolSet` would fetch VMs that correspond to a VMs pool rather than a VMSS.

And I would think that this function should check for both?

Reply:
This implementation falls back to the old IsDeleting logic for any non vmss pools due to urgency.